docs/sdk-and-tools/troubleshooting/rust-setup.md

```diff
@@ -0,0 +1,158 @@
+---
+id: rust-setup
+title: Fix Rust installation
+---
+
+[comment]: # (mx-abstract)
+
+When encountering issues with your Rust installation, we recommend a cleanup (uninstall) first, especially if you have multiple installations (by accident or on purpose).
+
+[comment]: # (mx-context-auto)
+
+## Uninstall existing Rust
+
+If you've installed Rust using your OS package manager:
+
+```bash
+# Ubuntu
+sudo apt remove cargo
+sudo apt autoremove
+```
+
+If you've installed Rust using `rustup`:
+
+```bash
+rustup self uninstall
+```
+
+If you've installed Rust using `mxpy` with a version older than `v9`:
+
+```bash
+rm -rf ~/multiversx-sdk/vendor-rust
+```
+
+:::note
+Since `mxpy v9` (November of 2023), `mxpy deps install rust` does not create an isolated Rust installation anymore in `~/multiversx-sdk/vendor-rust`. Instead, [it installs Rust _globally_](https://www.rust-lang.org/tools/install).
+:::
+
+If you've installed Rust using `mxpy v9` or later:
+
+```bash
+rustup self uninstall
+```
+
+[comment]: # (mx-context-auto)
+
+## Installing Rust and sc-meta
+
+[comment]: # (mx-context-auto)
+
+:::note
+`sc-meta` is universal smart contract management tool. Please follow [this](/developers/meta/sc-meta) for more information.
+:::
+
+### With mxpy
+
+```bash
+mxpy deps install rust --overwrite
+```
+
+For more information, go to [managing dependencies using `mxpy`](/sdk-and-tools/sdk-py/mxpy-cli/#managing-dependencies).
+
+[comment]: # (mx-context-auto)
+
+### Without mxpy
+
+As recommended on [rust-lang.org](https://www.rust-lang.org/tools/install):
+
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+Then, choose **Proceed with installation (default)**. 
+
+Once Rust is installed, switch to a nightly version and install the `wasm32-unknown-unknown` target:
+
+```bash
+rustup default nightly-2023-05-26
+rustup target add wasm32-unknown-unknown
+```
+
+Afterwards, install `sc-meta`:
```

Comment:
optional: maybe would be good to specify that when installing using **mxpy** `sc-meta`, `twiggy` and `wasm-opt` are also installed.

Reply:
Fixed.